---
title: "Queen pheromones have conserved effects on the transcriptome in ants and bees"
author:
- Luke Holman$^1$*, Heikki Helanterä$^{2}$, 
- Kalevi Trontti$^3$ and Alexander S. Mikheyev$^{4,5}$
- '*\textit{luke.holman@unimelb.edu.au} \vspace{5mm}'
output:
  pdf_document: 
    keep_tex: true
  html_document: default
bibliography: references.bib
fontsize: 12pt
header-includes: 
  \usepackage[all]{hypcap} 
  \usepackage{geometry}
  \geometry{a4paper, total={7in, 9in}}
  \usepackage{amsmath} 
  \usepackage{booktabs}
  \usepackage{caption}
  \usepackage[labelfont=bf]{caption} 
  \usepackage{titling} 
  \pretitle{\begin{flushleft}\LARGE}
  \posttitle{\par\end{flushleft}\vskip 0.5em} 
  \preauthor{\begin{flushleft}\large \lineskip 0.5em} 
  \postauthor{\par\end{flushleft}} 
  \predate{\begin{flushleft}\large} 
  \postdate{\par\end{flushleft}}
  \usepackage{fancyhdr} 
  \usepackage[left]{lineno}
  \linenumbers
  \pagestyle{fancy} 
  \fancyhead[LO,LE]{\textsl{\leftmark}} 
  \rhead[]{Transcriptomics of queen pheromones} 
  \usepackage[noabbrev,capitalise]{cleveref}
  \usepackage{titlefoot} 
  \usepackage{amssymb} 
  \usepackage{rotating}
  \usepackage{microtype}
csl: nature.csl
abstract: \noindent Queen pheromones are chemical signals that mediate reproductive division of labor in eusocial animals. Remarkably, queen pheromones are composed of identical or chemically similar compounds in some ants, wasps and bees, even though these taxa diverged >150MYA and evolved queens and workers independently. To test whether queen pheromones have similar effects across taxa, we measured the transcriptomic consequences of experimental exposure to queen pheromones in workers from two ant and two bee species (genera&#58; _Lasius_, _Apis_, _Bombus_). Queen pheromone exposure affected transcription and splicing at many loci. Many genes responded consistently in multiple species, and the set of pheromone-sensitive genes was enriched for functions relating to lipid biosynthesis and transport, olfaction, production of cuticle, oogenesis, and histone (de)acetylation. Pheromone-sensitive genes tended to be evolutionarily ancient, positively selected, peripheral in the gene coexpression network, hypomethylated, and caste-specific in their expression. Our results reveal how queen pheromones achieve their effects, and suggest that ants and bees use similar genetic modules to achieve reproductive division of labour. \vspace{5mm} \par\noindent \textbf{Keywords:} Fertility signal, Gene coexpression network analysis, Reproductive groundplan hypothesis, Sociogenomics. \vspace{5mm} \par\noindent \textbf{Word count:} 6057.
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

 \maketitle\unmarkedfntext{ 
$^1$School of BioSciences, University of Melbourne, Victoria, Australia.
$^2$Ecology and Genetics Research Unit, University of Oulu, Finland.
$^3$Department of Biosciences, University of Helsinki, Finland
$^4$Research School of Biology, Australian National University, Canberra, ACT 0200.
$^5$Ecology and evolution unit, Okinawa Institute of Science and Technology, Onna-son, Kunigami-gun, Okinawa, 904–0412, Japan.}
\newpage


# Introduction
Queen pheromones are chemical signals that characterise queens and other reproductive individuals in the social insects[@VanOystaeyen:2014vr; @Holman:2018tx]. These signals can affect the behaviour of other colony members, e.g. by attracting workers[@Slessor:1988cj], eliciting submissive behaviour[@Smith:2016dz], modulating aggression[@Smith:2009ic; @Yagound:2015ip], or inhibiting production of new queens[@Vargo:1988fv; @Winston:1990vi]. Queen pheromones also have long-lasting effects on individuals that encounter them, including reducing female fecundity[@VanOystaeyen:2014vr], influencing the rate at which workers progress to different tasks with age[@Pankiw:1998vta], and altering workers' capacity to learn[@Vergoz:2007ej]. Queen pheromones are regarded as an honest signal of fecundity and condition, to which workers adaptively respond by continuing to express a worker-like phenotype, as opposed to a 'manipulative' adaptation that reduces worker fitness[@Keller:1993vl; @Holman:2012ey; @Peso:2015hx; @Oi:2015go]. 

Current evidence suggests that most or all eusocial insects possess queen pheromones[@VanOystaeyen:2014vr; @Oi:2015go; @Holman:2018tx]. Although the eusocial bees, ants and wasps evolved eusociality (and thus, queen-worker communication) independently[@Peters:2017bk], most Hymenopteran queen pheromones are thought to be composed of chemically similar compounds[@VanOystaeyen:2014vr; @Holman:2018tx]. Certain species of ants, wasps and bumblebees have been experimentally shown to use cuticular hydrocarbons (CHCs; a non-volatile blend of hydrocarbons adhering to the body surface) as queen pheromones, particularly certain long-chained alkanes, methylalkanes, and alkenes[@VanOystaeyen:2014vr; @Holman:2018tx]. By contrast, honeybee queens (genus _Apis_) possess a pheromone composed of a blend of fatty-acid derived molecules (e.g. keto acids), which is secreted from the mandibular gland[@Slessor:1988cj]. The similarity of non-_Apis_ species’ queen pheromones implies that these pheromones evolved from chemical cues or signals (e.g. sex pheromones) that were already present in the non-social most recent common ancestor of the social Hymenoptera[@VanOystaeyen:2014vr; @Oi:2015go]. 

Presumably, the profound changes in worker behaviour and physiology caused by queen pheromones stem from pheromone-mediated effects on the transcriptome. For example, queen pheromone exposure might stimulate or repress the expression of transcription or splicing factors, or affect their ability to bind to promoter regions and splice sites (e.g. by modulating epigenetic processes[@Holman:2016ea]). To our knowledge, only two previous studies – both using microarrays in the honey bee _Apis mellifera_ – have experimentally measured the effects of queen pheromone on the whole transcriptome[@Grozinger:2003er; @Fussnecker:2011cg]. The response of other species to queen pheromones is unstudied, and so it is unclear whether different species use similar or distinct genetic pathways in queen-worker communication. The _Apis_ research also bears repeating because only 18 of _c_. 1000 differentially expressed genes from the first study replicated in the second (3-fold fewer than expected by chance[@Fussnecker:2011cg]). 

Here, we performed RNA sequencing on adult worker whole bodies to identify genes that are differentially expressed or alternatively spliced in response to experimental exposure to synthetic queen pheromones, in two bee and two ant species. Our first aim was to determine the extent to which pheromone-sensitive genes, pathways, and transcriptional modules are similar or distinct in ants and bees. The chemical similarity of some species’ queen pheromones, coupled with the fact that queen pheromones influence similar phenotypic traits across the Hymenoptera[@VanOystaeyen:2014vr], suggests that queen pheromones might affect many of the same genes across species. Conversely, we expect that some responses to pheromone will be unique to bees or to ants, given that these two taxa diverged over 150MYA, and independently evolved their eusocial societies (and thus, queen-worker communication). Second, we tested whether queen pheromones influence alternative splicing. Alternative splicing is thought to mediate many insect polyphenisms, including the queen-worker polyphenism[@Foret2012jf; @Klein:2016da; @Johnson:2016ko; @velasque2018doublesex], but is unstudied in relation to queen pheromones. Third, we aimed to identify genes and pathways that respond to queen pheromone, to reveal how these key social signals produce their manifold phenotypic effects. Fourth, we tested whether pheromone-deprived workers develop a more queen-like transcriptome[@Harrison:2015kf] to match their queen-like phenotype (e.g. laying eggs and living longer[@Dixon:2014it; @VanOystaeyen:2014vr]), thereby indirectly assisting the search for loci underlying caste dimorphism. Fifth, we tested whether the genes affected by queen pheromones tend to be older than eusociality itself, which is interesting in light of the theory that chemical signalling of fecundity provided an important stepping stone to the eusociality [@holman2014co].

# Results
## Effects of queen pheromone on gene expression
Many genes showed statistically significant differential expression between the pheromone-treated and control groups in _A. mellifera_ (322 genes), _L. flavus_ (290), and _L. niger_ (135), and a single gene was significant in _B. terrestris_ (Figure 1A; Tables S2-S5). The sets of significantly differentially expressed genes overlapped significantly more than expected for the two _Lasius_ species (Figure 1A; hypergeometric test: p < 0.0001). A smaller number of genes were significant in _Apis_ and one _Lasius_ species (Table S6), though the number of overlaps was not significantly higher than expected under the null (p = 0.19 for _L. flavus_ and p = 0.27 for _L. niger_). One gene was perturbed in 3/4 species: _myosin light chain alkali-like_ (Table S6). 

Venn diagrams like those in \autoref{fig1} can give a misleadingly low impression of the numbers of pheromone-sensitive genes, since all studies have finite power to detect differential expression for any particular gene. Moreover, having finite power causes one to underestimate the number of genes that overlap between species, because detecting overlaps requires one to avoid multiple false negatives. For example, if power were 40% per species to detect a particular conserved pheromone-sensitive gene, the chance to successfully detect the gene in all four species would be $0.4^4 = 2.6\%$.

![Summary of the effects of queen pheromone on gene expression and splicing, showing the extent of overlap between species. **A and B:** Venn diagrams showing the number of significantly differentially expressed or differentially spliced genes per species ($\alpha$ = 0.05 after false discovery rate correction). Parentheses in the outermost areas show this number as a percentage of all transcripts measured in the focal species, while parentheses in the inner areas show the number of overlapping genes as a percentage of the maximum number that _could_ have overlapped (this number depends on the number of detectable orthologs and the number of significant genes). Asterisks denote the one overlap that was significantly higher than expected by chance (hypergeometric test, p < 0.0001). _B. terrestris_ is omitted because there was only one significantly differentially expressed gene, and no differentially spliced genes. **C:** Graphical overview of the amount of similarity in pheromone sensitivity in the set of 3465 orthologous genes. The 4 inner rings show the pheromone sensitivity of each gene (redder colours indicate increased sensitivity), and the genes have been clustered according to coexpression pattern, as shown in the central dendrogram. The coloured outer ring shows the assignment of genes to modules (see Figure 3), and the grey area marked m0 refers to genes that were not assigned to a module. **D:** Orthologous genes tended to show a similar level of sensitivity to queen pheromones for each pair of species. The numbers give Spearman’s $\rho$ (p < $10^{-7}$ in all cases). \label{fig1}](../figures/MS/Figure 1 - revised.pdf)

For this reason, we employed additional, better-powered analyses to test for conserved effects across species (see Methods). Pheromone sensitivity was significantly positively correlated across pairs of orthologous genes, for all possible species pairs (Figures 1C and 1D; all p < $10^{-7}$). Thus, genes that were pheromone-sensitive in bees tended to also be pheromone-sensitive in ants. The cross-species correlations might be stronger than suggested by Figure 1D, because the sensitivity of each gene is measured with error, which would obscure any underlying correlation.

When we ranked orthologous genes in order of sensitivity to queen pheromone, there was some overlap between species in the top $n$ genes in the list (Table S8). For various $n$ (see Table S8), six genes appeared in the top $n$ most pheromone-sensitive genes for all four species: _serotonin receptor_, _protein takeout-like_, _titin-like_, _glucose dehydrogenase_, _histone-lysine N-methyltransferase SETMAR-like_, and _uncharacterized protein LOC102656088_. 

The gene showing the single largest change in expression in _A. mellifera_ was _Major Royal Jelly Protein 3_, which had 89-fold higher expression in workers deprived of queen pheromone (Table S2). Indeed, in _Apis_, five out of the top 12 most differentially expressed genes were _Major Royal Jelly Protein 1_, _2_, _3_, and _4_, plus _major royal jelly protein 3-like_. As well as being biologically interesting, these results provide a validity check: it is well-known that _Apis_ workers excrete royal jelly when rearing new queens, which they do if the queen (and her pheromone) are removed. Also, in _L. niger_, the second-most pheromone-sensitive gene was _Major Royal Jelly Protein 1_, which had 33-fold higher expression in controls (Table S5), suggesting that queen pheromones affect this gene family in ants as well as bees.



## Gene set enrichment analysis of highly pheromone-sensitive genes

Pheromone-sensitive genes (i.e. those showing a large difference in expression between treatments) were significantly enriched for many of the same GO (gene ontology) and KEGG (Kyoto Encyclopedia of Genes and Genomes) annotations across the four species (Figures 2 and S6-S9, and Tables S17-S20). For example, the GO: Biological process term _defense response to bacterium_ was significantly enriched in 3/4 species, and trended in the same direction for the fourth species (this result was driven by the pheromone sensitivity of genes like _defensin-1_, _hymenoptaecin_, and _apidaecin-1_). We also found 3/4 significant results for the GO: Molecular function terms _structural constituent of cuticle_ (driven by a large number of cuticular proteins), _odorant binding_, and _olfactory receptor activity_ (driven by many receptors and odorant binding proteins). Genes associated with the _extracellular region_ (driven by the major royal jelly proteins, the neuropeptide corazonin, and venom components) and the _plasma membrane_ (mostly olfaction-related) were similarly enriched among the pheromone-sensitive genes in 3/4 species. 

There was also cross-species conservation of several GO and KEGG terms related to fatty acid and amino acid biosynthesis (particularly synthesis of very-long-chain fatty acids, and unsaturated fatty acids, both of which are used in the synthesis of queen pheromone components), lipid transport (including _vitellogenin_), and the KEGG term _Neuroactive ligand-receptor intearctions_. Genes with the molecular function _sequence-specific DNA bind_ (i.e. transcription factors and the like) were also pheromone-sensitive. 



<!-- Many other terms were positively enriched in all four species, but only reached statistical significance for one. These terms are shown in the top rows of Figure 2, and include _RNA splicing_, _innate immune response_, _apoptotic process_, _Tryptophan metabolism_, and the _FoxO_ and _Hippo_ signalling pathways. Additionally, there were several terms pertaining to lipids, including  _Biosynthesis of unsaturated fatty acids_, _lipid transporter activity_, _Fatty acid elongation_, and various terms relating to the synthesis of 3-oxo and 3-ketoacyl fatty acid-CoAs. -->

<!-- A few terms related to the initiation of translation were positively enriched in both bee species and negatively enriched in both ants. Two other terms also showed this pattern; the pathway _Glycosphingolipid biosynthesis - ganglio series_, and the _MSL complex_ (male-specific lethal), which (among other things) mediates X-chromosome dosage compensation in _Drosophila_ and other non-Hymenopteran insects. -->



![\footnotesize Results of Gene Set Enrichment Analysis showing all Gene Ontology (GO) and Kyoto Encyclopedia of Genes and Genomes (KEGG) terms that were significantly overrepresented (red) or underrepresented (blue) among pheromone-sensitive genes in at least one of the four species. The colour shows the normalised expression score from gene set enrichment analysis. Asterisks denote statistically significant enrichment (p < 0.05), and double asterisks mark results that remained significant after Benjamini-Hochberg correction. Empty squares denote cases where we were unable to measure expression for at least 5 genes annotated with the focal term in one of the species. \label{fig2}](../figures/MS/Figure 2 - GO and KEGG.pdf)



<!-- For the Molecular Process ontology, pheromone-sensitive genes were significantly enriched for odorant binding and olfactory receptor activity in all 4 species, as expected given our manipulation of pheromone exposure. There was also enrichment of genes that produce structural constituents of the insect cuticle, lipid transporters including _vitellogenin_, steroid hormone receptors, as well as many cytochrome P450 genes involved in heme binding.  -->

<!-- For the Cellular Component ontology, we found that many pheromone-sensitive genes were membrane-bound (e.g. those involved in fatty acid biosynthesis), including to the post-synaptic membrane (e.g. nicotinic acetylcholine receptors). There was also enrichment of genes whose products are excreted to the extracellular region, such as the Major Royal Jelly Proteins. Genes related to microtubules and kinesin, as well as cellular respiration, were moderately enriched. -->

<!-- Analysis of annotated KEGG terms suggested that pheromone-sensitive genes were often involved in metabolism and biosynthesis, particularly of glycerolipids, long-chain unsaturated fatty acids, tyrosine, glycine, serine, and threonine. Queen pheromone also affected genes involved in neuroactive ligand-receptor interaction, and insect hormone biosynthesis.  -->


## Effects of queen pheromone on alternative splicing
Roughly 20% of genes had $\ge2$ detectable isoforms, in all four species (Figure S4), allowing us to test for pheromone-sensitive splicing. Pheromone treatment significantly elevated the expression of one isoform and repressed expression of another isoform for 52 genes in _A. mellifera_, 55 genes in _L. niger_, 22 genes in _L. flavus_, and no genes in _B. terrestris_ (\autoref{fig1}; Tables S10-S12). Three genes showed pheromone-sensitive splicing in more than one species, corresponding to around 4% of the maximum numbers of genes that could have overlapped (\autoref{fig1}). Again, these numbers could well be underestimates, since we have limited power to detect differential isoform expression, and each 'hit' requires two isoforms per gene to be statistically significant (i.e. we need 4 significant results to detect a single overlap between species). _DNA methyltransferase 3_ showed significantly pheromone-sensitive splicing in _L. niger_, recalling our previous qPCR result that queen pheromone affects DNA methyltransferase expression [@Holman:2016ea].

We next ranked all the alternatively-spliced genes with detectable orthologs in all four species in order of the sensitivity of their isoform profile to pheromone treatment (defined as the range in log fold change of the focal gene's isoforms), and performed gene set enrichment analysis on the resulting 'splicing sensitivity score'. Most GO and KEGG terms were enriched in similar patterns across species, e.g. _intracellular signal transduction_, _transmembrane transport_, _transcription, DNA-templated_, and _serine-type endopeptidase activity_ (Figure S6; Tables S21-S22). The GO terms showing non-significant trends toward enrichment included signal transduction, methyltransferase activity, mRNA processing, protein transport and modification, and microtubule motor activity. There was also a positive correlation between species in our measure of the sensitivity of splicing to pheromone treatment for all six possible species pairs, though only 2/6 of these correlations were significant (Table S21). The correlation was esepcially strong for the two bees ($\rho$ = 0.19, FDR-corrected p < $10^{-7}$), and was also signficant for _Bombus_ and _Lasius flavus_ ($\rho$ = 0.09, FDR-corrected p =	0.030). These results suggest that queen pheromone affects the splicing of some of the same loci across species.


## Pheromone-sensitive genes tend to pre-date the split between ants and bees
In all four species, the average pheromone sensitivity (i.e. absolute log fold difference between treatments) of "ancient" genes (i.e. those with a detectable ortholog in both ants and bees) was approximately double that of genes that are putatutively specific to bees or ants (Mann-Whitney tests, p < $10^{-15}$; Online Supplementary Material). This result suggests that most pheromone-sensitive genes existed prior to the evolutionary divergence of bees and ants, and thus pre-date the origin of eusociality.


## Effects of queen pheromone on the gene coexpression network
Among the 3465 genes for which orthologs were detected in all four species, we identified nine modules of coexpressed genes, each containing between 38 and 1639 genes; only 3% genes were left unassigned to a module (Figures 1C and 3; Tables S24-S33). The best-fitting multivariate model of the 9 modules' 'eigengenes' (a metric that quantifies the relative expression of entire modules; see Methods) contained Treatment as a predictor, but not Species or the Treatment $\times$ Species interaction (posterior model probability was >99%, indicating clear rejection of the Treatment $\times$ Species interaction, and the two models not containing Treatment; Table S13). This result suggests that some modules of coexpressed genes responded to pheromone treatment, and that the response is consistent across species. Specifically, modules 1, 4 and 9 showed a statistically significant difference in mean eigengenes between pheromone treatments (Figures 1C and 3; Table S14). 

![\footnotesize Boxplots showing the distribution of ‘eigengenes’ across samples for each of the nine transcriptional modules identified via weighted gene coexpression analysis. The eigengene is a measure of the expression level of a transcriptional module, relative to other samples in the set. Queen pheromone treatment had a statistically significant effect across species for modules 1, 4 and 9 (the annotations give Cohen's d effect size and its 95% CIs, estimated from a multivariate Bayesian model of all nine modules).  \label{fig3}](../figures/MS/Figure 3 - Module eigengenes.pdf)

The pheromone-sensitive module 1 was large (1639 genes), and was enriched for GO and KEGG terms related to the cell cycle, DNA repair, transcription and splicing of RNA, and ribosomes (Figures 4 and S7-S9; Table S23). Module 1 also contained genes relating to the epigenome, such as _DNA methyltransferase 3_ and several histone deacetylases and methyltransferases. Module 4 (288 genes) was enriched for GO terms relating to the pentose phosphate pathway, fatty acid and amino acid biosynthesis, lipid metabolism, vesicle-mediated transport, and for genes associated with the endoplasmic reticulum (where proteins, lipids, and steroid horomones are made). This module also contained genes for synthesising very long-chained fatty acids and acetyl-CoA, which are precursor substances for the synthesis of cuticular hydrocarbons[@Blomquist:2010ux; @Dembeck:2015jb] and the main components of honeybee queen mandibular pheromone[@Wu:2017br]. Module 9 was enriched for purine metabolism; purines are required for cell division and transcription, and to produce important biomolecules like ATP, NADH, and coenzyme A.

![Results of KEGG pathway enrichment analysis for the genes in each of the three significantly pheromone-sensitive transcriptional modules. The gene universe was defined as all genes for which we found an ortholog in all four species (i.e. the set that was used to discover these co-expressed modules). All KEGG terms shown in green were significantly enriched (p < 0.05), and those shown in purple remained significant after correction for multiple testing. Fold enrichment was calculated as the proportion of genes associated with the focal KEGG term in the module, divided by the equivalent proportion in the gene universe. Figures S7-S9 show equivalent GO enrichment results.  \label{fig4}](../figures/MS/Figure 4 - module KEGG.pdf)


## Pheromone-sensitive genes have low connectedness
We found a negative correlation between sensitivity to queen pheromone and connectedness across genes (Spearman’s $\rho$ > 0.24, p < $10^{-48}$ for all species). This means that highly pheromone-sensitive genes are expressed comparatively independently of the rest of the transcriptome, while highly connected genes tend to be insensitive to queen pheromone. This result is illustrated by the excess of pheromone-sensitive genes in Module 0 (which holds the few genes that were expressed relatively independently of Modules 1-9) in Figure 1C. 


## Characteristics of pheromone-sensitive genes in _Apis mellifera_
\autoref{fig5} summarises the correlations across genes for a number of gene-level properties, for honeybees. On average, strongly pheromone-sensitive genes had less gene body DNA methylation, lower expression levels, and lower codon usage bias. Pheromone-sensitive genes had higher values of $\gamma$, meaning that they have been under stronger positive selection and/or weaker purifying selection[@Harpur:2014hk]. We also found a positive relationship between pheromone sensitivity and the extent to which a gene was upregulated in queens relative to sterile workers (as measured in [@Grozinger:2007ki]). We did not find a significant correlation between a gene's pheromone sensitivity and the caste-specificity of its histone modifications (averaged across the gene, using published ChIPseq data[@wojciechowski2018ph]). However, almost all variables were strongly inter-correlated (\autoref{fig5}), making the causal relationships among them (if any) difficult to infer without further evidence.

![\footnotesize Spearman correlations for various gene-level measurements from the present study and earlier research, for _Apis mellifera_ (measurements from the present study are shown in bold). 'Pheromone sensitivity' was calculated as the absolute value of the Log$_2$ fold difference in expression between pheromone treatment and the control. Expression level shows the logarithm of the average across our 6 _Apis_ libraries. For the 'Upregulation in queens/fertile workers' data[@Grozinger:2003er], positive values denote genes that have higher expression in queens or fertile workers, relative to sterile workers. For the three histone modification variables[@wojciechowski2018ph], high values indicate that the modification is more abundant in queen-destined larvae, and low values indicate it is more abundant in worker-destined larvae. The two DNA methylation variables give two different measures of the amount of gene body DNA methylation, namely an indirect measure (-log CpG O/E ratio) and a direct measure (BiS-seq[@Galbraith:2016gz]). Codon usage bias was estimated using the codon adaptation index: high values indicate bias for particular synonymous codons. Lastly, the parameter gamma ($\gamma$) describes the form of selection, where positive values denote positive selection, and negative values purifying selection[@Harpur:2014hk]. The asterisks indicate the p-values after FDR correction (* p < 0.05; ** p < 0.01; *** p < 0.0001). \label{fig5}](../figures/MS/Figure 5 - correlations.pdf)

## Comparison with caste-specific gene expression in ants
Hypergeometric tests revealed six instances in which one of our gene co-expression modules overlapped significantly with one of the modules from Morandin et al.’s study[@Morandin:2016jd] of caste-biased gene expression in ants, after correcting for multiple testing (Table S16). Modules 2, 3, and 8 from our study overlapped with worker-biased modules, and Modules 1 and 4 overlapped with queen-biased modules. Since Modules 1 and 4 are pheromone-sensitive (\autoref{fig3}), these results suggest that the set of pheromone-sensitive genes overlaps with the set of caste-biased genes, in ants (as well as bees; \autoref{fig5}). Ten genes were found in both our Module 4 and Morandin et al.’s queen-biased module (Table S14); these genes included _protein takeout-like_, _NAD kinase 2, mitochondrial-like_, _histone H2A-like_ and _methyltransferase-like_, again implicating _takeout_, metabolism, and epigenetic processes in caste polyphenism and the response to queen pheromone. 


# Discussion

To a first approximation, queen pheromones had similar effects on the transcriptome in all four species. For example, orthologous genes in bees and ants tended to show similar levels of pheromone sensitivity, and we identified three transcriptional modules showing a consistent response to queen pheromone across species. Accordingly, gene set enrichment analysis revealed that broadly similar functional categories of genes were enriched in bees and ants. This cross-family conservation is not unexpected, given that queen pheromones induce similar phenotypes (e.g. sterility) in both taxa. However, this outcome was not a foregone conclusion, for example because bees and ants evolved eusociality independently (and have few caste-specific genes in common[@Berens:2015hg]), and because bumblebees have smaller, annual colonies in which behavioral interactions play a larger role in regulating reproductive division of labour [@doorn1989fa]. 

In _Apis_ and both _Lasius_ species, we found that queen pheromone treatment caused statistically significant changes in alternative splicing at multiple loci, increasing the expression of certain isoforms while inhibiting that of others. The lack of a significant effect in _B. terrestris_ might well be a false negative, since the estimated sensitivity of alternative splicing to queen pheromone was strongly correlated across pairs of orthologous _Apis_ and _Bombus_ genes. Also, _Bombus_ genes with high (though non-significant) pheromone-sensitive splicing were significantly enriched for similar GO and KEGG terms as those in the other three species (Figure S6). Our study thus adds to the growing list of cases in which alternative splicing underlies polyphenisms in insects [@kijimoto2012div; @wang2014lo; @zhuo2017tra; @velasque2018doublesex].

In _Apis_, pheromone-sensitive genes tended to be positively selected, weakly connected in the transcriptional network, weakly expressed, and hypomethylated, relative to pheromone-insensitive genes. Additionally, queen pheromones affected a somewhat similar set of genes to that which distinguishes adult queens and workers in bees as well as ants, consistent with our prediction that queen pheromones would make gene expression more worker-like. Finally, pheromone-sensitive genes were disproportionately likely to pre-date the divergence of bees and ants, consistent with their patterns of enrichment for GO and KEGG terms associated with taxonomically ubiquitous processes such as lipid biosynthesis. 

Many genes or gene families were differentially expressed in two or more species. As one example, queen pheromone inhibited the expression of major royal jelly proteins (MRJPs) in honeybees (echoing earlier findings that MRJPs expression covaries with reproductive physiology[@thompson2008ge]) and in _L. niger_. Among other functions, MRJPs are essential for rearing new queens, which workers do when their current queen dies or becomes infertile (i.e. when the queen pheromone disappears)[@Winston:1990vi]. MRJPs are produced during development and in the adult fat body, and belong to the phylogenetically ancient _yellow_ gene family[@Drapeau:2006jt], which has diverse roles in development, the nervous system, behaviour, immunity, and pigmentation. 

Genes related to synthesis and transport of lipids and fatty acids formed a strongly co-expressed transcriptional module, which was modulated by queen pheromone across taxa. The affected genes included enzymes for making long-chained fatty acids and fatty acyl-CoAs, which are biosynthetic precursors of cuticular hydrocarbons (CHCs) as well as the components of the queen mandibular pheromone (QMP) of honeybees[@Blomquist:2010ux; @Dembeck:2015jb; @Wu:2017br]. Additionally, a number of genes putatively involved in CHC and QMP biosynthesis, such as cytochrome P450s, NADPH synthases, and genes involved in fatty acid elongation and oxidoreductase activity[@hasegawa2009di; @Blomquist:2010ux; @Dembeck:2015jb; @Wu:2017br], were differentially expressed. We also observed large (though non-significant) effects of queen pheromone on the expression of _vitellogenin_ (a lipid transporter) and _hexamerin 70a precursor_, two classic 'eusociality genes' that have been linked to caste and oogenesis by many previous studies (e.g.[@Nelson:2007fz; @Martins:2011co]). These results are expected given that pheromone-deprived workers begin depositing yolk in their ovaries via lipid synthesis and transport[@CruzLandim:2006uz]. 

Our results hint at the mechanism by which queen pheromones achieve their effects, and suggest a novel (and heretofor missing[@Holman:2012ey; @Oi:2015go]) mechanism underlying the widely-observed honest signalling of fecundity via olfactory cues/signals in social insects. This honest signalling is considered a puzzle because of the apparent fitness benefits of exaggerating one's fecundity via pheromones (in queens) or of 'covert' reproduction without pheromonal signalling (in workers)[@Smith:2009ic; @Holman:2012ey; @Holman:2013iy; @VanOystaeyen:2014vr; @Oi:2015go]. We speculate that the fatty acid-derived queen pheromones found in ants and bees are absorbed directly into the body (e.g. by ingestion), where they inhibit lipid biosynthesis via negative feedback, thereby inhibiting oogenesis. If this hypothesis proves correct, the colony could be regarded as having a shared 'social physiology', whereby colony members keep track of their own physiological state via standard within-body signals (e.g. juvenile hormone, insulin signalling), as well of the states of other individuals via pheromones[@johnson2010de]. We also speculate that workers evolved elevated sensitivity to queen pheromones as colony sizes increased over evolutionary time, e.g. via changes in olfaction and physiology, allowing them to continue to express the sterility response even though their contact with the queen is reduced. Lastly, the necessity of lipid synthesis and transport for oogenesis, coupled with an inextricable, non-evolvable link between the makeup of the internal and external lipid profiles, would enforce a reliable correlation between individual fecundity and odour profile[@Smith:2009ic; @Holman:2012ey]. 

In another notable result, the gene _protein takeout-like_ was among the most strongly pheromone-sensitive genes in all four species. The _takeout_ family encodes proteins that are expressed in, and secreted from, the brain-associated fat body and antennae, and some members putatively bind juvenile hormone[@Hagai:cp]. Interestingly, _takeout_ genes have been linked to discrete polyphenisms in termites[@Hojo:2005ht], locusts[@Guo:2011ks] and aphids[@Ghanim:2006bb], suggesting that they might be similarly important to polymorphism in the eusocial Hymenoptera. Additionally, in _Drosophila_, the expression of _takeout_ is stimulated by the male-typical isoforms of the sex differentiation genes _doublesex_ and _fruitless_, and suppressed by the female isoforms[@Dauwalder:2002ga]. This is noteworthy in light of the recently-proposed hypothesis that sex differentiation genes such as _doublesex_ have been co-opted to control caste polyphenism in eusocial insects[@Klein:2016da; @Johnson:2016ko; @velasque2018doublesex]. This finding also brings to mind the 'reproductive groundplan hypothesis'[@johnson2010de], which posits that reproductive division of labour is the result of regulatory evolution of nutrition signalling pathways, e.g. insulin-like signalling, which (among other things) controls the balances of lipid and sugar synthesis and metabolism, and the rate of ageing (which differs between queens and workers, and between fertile and sterile workers[@Dixon:2014it]).

Additionally, _serotonin receptor_ was among the most pheromone-sensitive genes in all four species. Serotonin seems understudied in social insects, although two studies have found differences in serotonin titre or receptor gene expression between sterile and fertile workers, in _Polistes_ wasps [@sasaki2007ne] and _Apis_ [@vergoz2012bio]. Another biogenic amine -– dopamine -– is better-studied; it has been implicated in the response to queen pheromone in _Apis_[@Beggs:2007dm], and affects behaviour and fecundity in many insects[@McQuillan:2012ir; @Okada:2015cg; @Weitekamp:2017if]. There was also some evidence that the expression of the neurohormone corazonin was modulated by queen pheromone (e.g. Table S17), consistent with experimental results showing that corazonin induces worker-like behaviors and suppresses queen-like behaviors in an ant [@gospocic2017ne].

Several genes related to myosin, which functions in muscle contraction, were significantly downregulated in the queen pheromone treatment in at least one species. Interestingly, a recent study compared the transcriptomes of queens and workers in 16 ant species with RNA-seq, and found only a single gene that was significantly differentially expressed between castes in all 16 species: another myosin gene[@Morandin:2016jd]. Myosin genes are also differentially expressed between fertile and sterile workers[@Cardoen:2011dn; @Niu:2014gu] and queen- and worker- destined larvae[@Barchuk:2007vs] in honeybees, and between queens and workers in bumblebees[@Harrison:2015kf]. We speculate that myosins show caste-specific expression due to caste differences in muscle morphology and activity levels (e.g. queen ants fly while workers do not, and in bees, there is a caste difference in flight frequency). 

The consistently pheromone-sensitive module 1 contained many genes relating to histone modification, particularly histone-lysine N-methyltransferases and histone deacetylases. These included _histone-lysine N-methyltransferase eggless_, which trimethylates Lys-9 of histone H3 in the _Drosophila_ ovary, and which is essential for oogenesis (FlyBase: FBgn0086908). Another interesting gene was _male-specific lethal 1 homolog_, which regulates gene expression by acetlyation of H4 lysine 16; the resulting H4K16ac 'epimark' is hypothesised to regulate the development and renewal of female germline stem cells[@sun2015his]. Another histone acetlyation epimark, H3K27ac, is related to the major-minor worker size polymorphism in carpenter ants[@simola2016epi], and differs between queens and workers in honeybees[@wojciechowski2018ph]. In sum, it seems likely that the receipt of queen pheromone causes a rewiring of the epigenome, which in turn regulates the genes underlying oogenesis. We also found that queen pheromone affected the splicing of _DNA methyltransferase 3_ (_dnmt3_) in _Lasius niger_, echoing our previous work showing that queen pheromones affect the expression of _dnmt1_ and _dnmt3_ in bees and ants[@Holman:2016ea], and again paralleling evidence that differential DNA methylation is involved in queen-worker polyphenism[@Kucharski:2008gu]. Direct measurement of the effect of queen pheromone on the epigenome has not yet been performed (but see[@Holman:2016ea]).

A number of recent papers on the origins of eusociality have asked whether the key genetic players tend to be ‘ancient’ genes with fundamental cellular functions, or more recently-evolved genes with specialised functions (e.g.[@Patalano:2015vl; @Morandin:2016jd; @Berens:2015hg]). Most of this work has focused on genes showing queen- or worker-biased expression, but since that gene set overlaps substantially with the set of pheromone-sensitive genes, our results are apposite. We found that pheromone-sensitive genes tend to predate the split between bees and ants, suggesting that present-day queen pheromones primarily affect genes that already existed in the genomes of the first eusocial insects. However, we also found that pheromone-sensitive genes had low connectedness, expression levels, and codon usage bias; none of these characteristics are consistent with the targets of queen pheromone being 'housekeeping' genes, i.e. extremely ancient, constitutively-expressed genes with ubiquitous cellular functions[@Morandin:2016jd]. Instead, queen pheromone affected a moderately-sized subset of the transcriptome, whose expression varied relatively independently of the remainder. This result is interesting because genetic modules showing flexible expression patterns and reduced pleiotropy are predicted to be major drivers of adaptation because they are comparatively free to undergo adaptation[@Gerhart:2007eq]. Our results are consistent with a model whereby a relatively self-contained genetic module (controlling nutrient homeostasis, and thus oogenesis) acquired a new expression pattern, producing the observed polyphenism of fertile and sterile females. Subsequently, the genes in this module underwent adaptation to their new roles, explaining our result that pheromone-sensitive genes are both evolutionarily ancient and positively selected. 



<!-- Several results suggest that queen pheromones affect the central nervous system in bees and ants. For example, genes whose products localise to the post-synaptic membrane were significantly enriched in all four species, as were genes annotated with the KEGG term _neuroactive ligand-receptor interaction_. Additionally,  -->


<!-- Juvenile hormone affects fecundity, longevity, and queen pheromone production in ants, bees and wasps[@Holman:2012ey; @Robinson:1997vq; @Oliveira:2017jx], and queen pheromone affects juvenile hormone titre in honeybees[@Pankiw:1998vta]. Our results thus reaffirm previous conclusions that juvenile hormone is involved in the response to queen pheromone[@Pankiw:1998vta].  -->




# Methods
## RNA sequencing of pheromone-treated bees and ants
The present study uses RNA obtained from the same insect samples as those used in an earlier study[@Holman:2016ea], which provides complete methods for the pheromone bioassay, RNA extraction, and preparation of cDNA. Briefly, we treated nest boxes containing workers from 3-8 colonies per species with either a solvent-only control or their own species’ queen pheromone, and then extracted total RNA from individual workers (either whole bodies, or a random lateral body half for _Bombus_), removed genomic DNA with DNase, and reverse-transcribed RNA to cDNA. For _Apis mellifera_ honeybees, the pheromone used was commercially available Queen Mandibular Pheromone (QMP), which is a mixture of 5 chemicals (principally keto acids). For _Bombus terrestris_, the pheromone was the CHC pentacosane (C$_{25}$), and for the two _Lasius_ ant species it was the CHC 3-methylhentriacontane (3-MeC$_{31}$). The _B. terrestris_ workers were from the same cohort and colonies as in Holman [-@Holman:2014jq], though they were different individuals.

We then used Qubit fluorometry to determine the mass of cDNA obtained from each worker, and pooled equal amounts of cDNA from five randomly-selected workers for each combination of species, colony, and treatment. Not including 4 problematic samples that were later discarded (see Figures S1-S2), we sequenced 44 cDNA pools (6 for _A. mellifera_, 10 for _B. terrestris_, 13 for _L. flavus_, and 10 for _L. niger_; Table S1); library preparation (using Illumina TruSeq kits) and sequencing was conducted by Edinburgh Genomics. The libraries were sequenced in three lanes of an Illumina HiSeq 2500 sequencer set to high output, yielding 125bp paired-end reads. All samples were individually barcoded and run in all three lanes, preventing lane effects from confounding the experiment. The experiment yielded 14±1.3 (st. dev.), 12.3±4.3, 14.7±2.7, 13.1±1.4 million reads for _A. mellifera_, _B. terrestris_, _L. flavus_ and _L. niger_, respectively. We used Trimmomatic[@Bolger:2014ek] to remove sequencing primers, and trimmed reads for quality using the SLIDINGWINDOW:4:15 parameter prior to subsequent analyses. After trimming, the number of paired reads was 10.8±1, 8.3±2.9, 10.2±1.4, 10.8±1.3 million, respectively.

## Quantifying differences in gene expression or alternative splicing between pheromone treatments
We aligned and quantified the raw reads using the RSEM[@Li:2011cb] pipeline with Bowtie2[@Langmead:2012ka] to transcripts from published genomes for _A. mellifera_, _B. terrestris_ and _L. niger_[@Consortium:2006wj; @BenMSadd:2015bq; @Konorov:2017ft].  The _L. niger_ genome assembly had no isoform information, so we identified isoforms using Tophat[@Trapnell:2009dp]. No reference genome was available for _L. flavus_, so we assembled the transcriptome _de novo_ using Trinity[@Grabherr:2011jb], and identified coding regions with TransDecoder[@Haas:2013jq]. 

Within each species, we used EBSeq-HMM[@Leng:2015dw] to calculate the fold difference in expression between the control and pheromone-treated workers for each transcript using the rsem-run-ebseq pipeline implemented in Trinity. We adjusted p-values to control the false discovery rate using the Benjamini-Hochberg method, then defined genes with adjusted p<0.05 as significantly differentially expressed. As a sensitivity analysis, we re-ran the EBSeq analysis after removing low-abundance transcripts, increasing power by reducing the number of tests; we obtained essentially identical lists of significant genes to those from the full analysis. 

To identify genes whose splicing was significantly affected by queen pheromone, we searched for genes for which at least one isoform was significantly upregulated in the control, while another isoform was significantly downregulated. We also calculated a 'pheromone sensitivity of splicing' score, by taking the maximum difference in log fold change for the isoforms of each gene (e.g. a gene with three isoforms showing -2, +0.1, and +1 log fold difference between treatments would score 3). This score was use to test for correlations across species in the gene-level sensitivity of splicing to pheromone, and for GO and KEGG enrichment tests of pheromone-sensitive splicing (see below).

## Testing for conserved effects of queen pheromone across species
The simplest method to identify conserved effects on gene expression is to tally the number of orthologous genes showing significant differential expression in two or more species (as in the Venn diagrams in Figure 1). Though robust, this method is highly conservative, because one has finite statistical power to detect any given differentially expressed gene. Power issues are compounded when searching for genes that show a conserved response across species, since one needs to avoid two or more false negatives per locus. We therefore performed two additional formal analyses to test for conserved transcriptional effects of queen pheromones, as well as plotting the pheromone sensitivity for each gene (Figure 1C) to allow qualitative assessment of the extent of cross-species similarity.

For the first formal test, we tested whether the pheromone sensitivity of each gene is correlated in each pair of species, using Spearman’s rank correlation on pairs of orthologous genes (see Figures 1C; 1D). Pheromone sensitivity was defined as the absolute log fold difference in expression between treatments. This test has improved power relative to the Venn diagram appraoch, but does not reveal the number or identity of the conserved/convergent pheromone-sensitive genes.

Secondly, we identified the genes that had detectable orthologs in all four species (defining orthologs as genes that were each other’s best BLAST hit, with both e-values $<10^{-4}$), and then ranked them from most to least pheromone-sensitive within each species. Then, we asked which genes appeared in the top $n$-most pheromone sensitive genes in all four species, for $n$=100, 200... 500. This analysis has good power to identify candidate genes that responded to pheromones in all four species, but runs the risk of false positives (i.e. genes that topped all four gene lists by chance alone). 


## Evolutionary age of pheromone-sensitive genes 
To test whether pheromone-sensitive genes tend to have an ancient or recent evolutionarily origin, we classified genes as either 'ancient' or 'putatively family-specific' using reciprocal best BLAST. Bee genes (_Apis_ or _Bombus_) with a BLAST hit (e-value $10^{-4}$) in at least one of the ant species were classified as ancient, and _vice versa_. Genes that were not classified as ancient might be false negatives (e.g. due to gaps in our sequence data, or because genes were lost in one lineage), hence our caution in labelling genes as family-specific. Any misclassifications should make it harder to detect a difference between pheromone-sensitive and -insensitive genes, but could not produce a spurious difference.

## Gene co-expression network analysis
We constructed a gene co-expression network for all four species, which included all genes for which orthologs were detected in all species, following Morandin et al.[-@Morandin:2016jd]. The aim of this analysis was to search for 'modules' of co-expressed genes that change their expression in response to queen pheromone in all the species. We therefore used an empirical Bayes method[@johnson:2007adj] (implemented via the ComBat function in R's _sva_ package[@Leek:2012ee]) to transform the expression data so as to remove multivariate differences in expression attributable to species or colony, clarifying the effect of pheromone treatment on the transcriptome. 

We used the R package WGCNA[@Langfelder:2008bd] to define the gene co-expression network and identify transcriptional modules, largely using the default settings. The two exceptions were that we imposed a minimum size for transcriptional modules of 30 genes, and used a signed (rather than unsigned) coexpression network. These choices mean that our analysis recovers modules of 30+ genes that are all simultaneously up- or down-regulated across our 44 samples. 

To test whether species, treatment, and their interaction explained variation in module 'eigengenes' (a metric describing the expression level of a particular module in the focal sample, relative to the other samples[@Langfelder:2008bd]), we used Bayesian multivariate models implemented in the R package `brms` [@burkner2017brms]. We fit five candidate models, all with the 9 eigengenes as a multivariate response, colony as a random effect, and Gaussian errors. The five models differed in their fixed effects, and we compared the models' fits in order to test for significant effects of treatment, species, and their interaction (using posterior model probabilities, calculated using bridge sampling). 

We also used the co-expression network to calculate connectedness for all genes. We defined connectedness as the sum of the correlations in expression between a given gene and every other gene in the network[@Masalia:2017ki]. Thus, a 'highly-connected gene' is one whose expression varies in concert with many other genes, across samples.

## GO and KEGG enrichment analyses
We downloaded Kyoto Encyclopedia of Genes and Genomes (KEGG) from the KEGG API and gene ontology (GO) terms from NCBI, for the best-annotated of our four species, _Apis mellifera_. KEGG terms group together genes that are known to interact in biochemical pathways, while GO classifies genes by Biological Process, Molecular Function, or Cellular Component. Genes in non-_Apis_ species were assumed to have the same GO and KEGG terms as their reciprocal best BLAST hits in _A. mellifera_. 

We implemented Kolmogorov-Smirnov enrichment tests (also called Gene Set Enrichment Analysis or GSEA[@subramanian2005gene]) using the `fgsea` package for R. These tests rank all genes in the set under test (called the 'gene universe') in order of some metric of interest, and then identify GO or KEGG terms that are significantly over- and under-represented among the top-ranked genes, relative to bootstrapped random expectations. As well as presenting the uncorrected p-values, we corrected the GO and KEGG results for multiple testing using the Benjamini-Hochberg method, though we note that this approach is crude and probably overly-conservative, since tests of the different terms are not independent. GO results were simplified by collapsing redundant GO terms into higher-order ones using the `collapsePathways` function in `fgsea`.

To identify enriched GO and KEGG terms among genes whose expression was sensitive to pheromone treatment, we ranked genes by the log_{10} posterior probability of differential expression (computed by EBSeq-HMM) and defined the gene universe as all genes (for _Apis_) or all genes with a detectable ortholog in _Apis_ (for other species). To identify enriched terms among genes with pheromone-sensitive splicing, we ranked genes by their splicing score, and specified the gene universe as all alternatively-spliced genes with _Apis_ orthologs.

To identify enriched GO and KEGG terms for each of the 9 co-expressed genetic modules, we used standard hypergeometric tests (implemented in the `clusterProfiler` R package), and defined the gene universe as all 3465 genes used in the coexpression network analysis. 


## Characteristics of pheromone-sensitive genes in _Apis_
_Apis mellifera_ honeybees are well-studied relative to our other species, and so we compared our pheromone sensitivity and connectedness data to pre-existing gene-level data from _A. mellifera_ using Spearman correlations.

First, we used published microarray results[@Grozinger:2007ki] to test whether pheromone-sensitive genes also showed a large difference in expression between A) queens and sterile workers, and B) fertile workers and sterile workers. Second, we examined codon usage bias, as measured by the codon adaptation index[@Sharp:1987tz]; high values indicate a bias towards particular synonymous codons in the coding regions of a gene. Third, we tested for relationships with the frequency of DNA methylation within the gene body, using two complementary measures of DNA methylation: the amount of CpG depletion (measured as the negative log observed/expected CpG ratio), or the percentage of methylated cytosines, estimated using whole genome bisulphite sequencing[@Galbraith:2016gz]. Fourth, we tested whether pheromone-sensitive genes show signatures of positive or purifying selection since the split between _A. mellifera_ and its congeneric _A. cerana_, using the metric $\gamma$[@Harpur:2014hk]. Lastly, we tested whether pheromone-sensitivity was correlated with the log expression level of each gene, using the average expression levels from the present study.

## Comparison with caste-specific gene expression in ants
Using reciprocal best BLAST (e-value $10^{-4}$), we attempted to classify the groups of orthologous genes from our study into one of the orthologous gene groups defined for queens and workers from 16 ant species in Morandin et al.[-@Morandin:2016jd]. We then tested for significant overlap between that study’s 36 gene co-expression modules, and the modules from our own study, using hypergeometric tests on all possible pairs of modules (followed by FDR correction). We thereby tested whether the groups of coexpressed genes that respond to pheromone also tend to show differential expression between queens and workers.

## Data availability and reproducibility
The sequencing data have been deposited at NCBI (BioSample ascensions: SAMD00106316-58). Bash, Python, and R scripts used to reproduce our bioinformatics pipeline and data analysis are archived at Github (https://github.com/mikheyev/queen-pheromone). As well as the supplementary figures and tables, our Online Supplementary Material contains the R scripts used to generate each result (the supplement can also be viewed online: https://mikheyev.github.io/queen-pheromone).

# Acknowledgements	
We are very grateful to Lijun Qiu, Tapio Envall, Sini Vuorensyrjä, Heini Ali-Kovero and Minna Tuominen for laboratory assistance, to Soojin Yi and Brendan Hunt for sharing data, and to Mark and Kirsten Holman for beekeeping. Brian Hanley and Jocelyn Millar kindly provided synthetic 3-MeC$_{31}$. Claire Morandin and Tim Linksvayer provided helpful discussion. 

# Funding
This work received funding from the Research School of Biology at Australian National University to LH; a Discovery Project (DP170100772) to LH and ASM; the Kone Foundation to HH; the Academy of Finland to HH (135970, 127390), and the Center of Excellence in Biological Interactions (284666).

# References
<div id="refs"></div>
